### PR TITLE
Enable the user to support hold foreign units.

### PR DESCRIPTION
**Pull request by bryancodazen**
### Summary
This PR enables users to support hold a foreign unit. 

To support hold a foreign unit the user must do the following:
1. double click the supporting unit
2. single click the foreign unit being supported
3. single click the territory the foreign unit is on

### Video
https://drive.google.com/file/d/1XUCNQGrxfFf5KfprwJaGfSVu6Pa4EJ5V/view?usp=sharing

### CODE REVIEW

File: beta-src/src/components/ui/WDUnitController.tsx [hunk 1]
```diff
@@ -54,23 +52,7 @@ const WDUnitController: React.FC<UnitControllerProps> = function ({
...
-      for (let i = 0; i < currentOrders.length; i += 1) {
-        if (currentOrders[i].unitID === meta.unit.id) {
-          unitCanInitiateOrder = true;
-          break;
-        }
-      }
-    }
-  }
-
   const clickAction = (e) => {
```
Comment (mariomelchor) [inline beta-src/src/components/ui/WDUnitController.tsx:55]: both clickAction and doubleClickAction are nearly identical now only thing that change is the method. Can we consolidate and pass another parameter to this method.

File: beta-src/src/state/game/game-api-slice.ts [hunk 2]
```diff
@@ -344,63 +344,67 @@ const gameApiSlice = createSlice({
...
       if (inProgress) {
-        if (order.type === "hold" && order.onTerritory !== null) {
+        if (unitID === clickData.payload.unitID) {
+          resetOrder(state);
+        } else if (type === "hold" && onTerritory !== null) {
           highlightMapTerritoriesBasedOnStatuses(state);
-        } else if (order.type === "move" && order.toTerritory !== null) {
+        } else if (type === "move" && toTerritory !== null) {
           highlightMapTerritoriesBasedOnStatuses(state);
         } else if (
```
Comment (mariomelchor) [inline beta-src/src/state/game/game-api-slice.ts:392]: This else if statement is doing the same thing for both checks. Can we do an `||` statement in the first check to include `type === "move" && toTerritory !== null`

**Review by mariomelchor — CHANGES_REQUESTED** [review]

**Review by mariomelchor — APPROVED** [review]
LGTM

**Review by sariacodazen — APPROVED** [review]
LGTM

**Pull request merged**